### PR TITLE
Added callback support

### PR DESCRIPTION
**Pull request by fwoelffel**
Also improved the documentation (+some refactoring)

### CODE REVIEW

File: pi-blaster.js [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
+ * @param {Function} callback Must accept only one optional error parameter
  */
-function writeCommand(cmd) {
+function writeCommand(cmd, callback) {
 	var buffer = new Buffer(cmd + "\n");
-	
 	var fd = fs.open(PI_BLASTER_PATH, "w", undefined, function(err, fd) {
 		if (err)
-			console.log("Error opening file: " + err);
+			callback(err);
```
Comment (sarfata) [inline pi-blaster.js]: I think you should check that `callback` is defined here, this way `writeCommand()` will remain backwards compatible and will work even if the caller does not provide a callback.

File: pi-blaster.js [hunk 2]
```diff
@@ -1,40 +1,46 @@
...
 		else {
 			fs.write(fd, buffer, 0, buffer.length, -1, function(error, written, buffer) {
-				if (error) 
-					console.log("Error occured writing to " + PI_BLASTER_PATH + ": " + error);
-				else
+				if (error)
+					callback(error);
```
Comment (sarfata) [inline pi-blaster.js]: same here

**Comment by sarfata** [comment]
Thanks for cleaning up my JS ;) Would you like to add a list of contributors in the readme and put your name in there?

**Comment by fwoelffel** [comment]
My bad, I didn't thought about the backward compatibility. It's fixed now ;)

**Comment by sarfata** [comment]
Thanks! Will bump the version and publish to npm right now.

**Pull request merged**